src/master-playlist-controller.js

```diff
@@ -318,7 +352,12 @@ export class MasterPlaylistController extends videojs.EventTarget {
           selectedMedia = this.selectPlaylist();
         }
 
+        if (!selectedMedia || !this.shouldSwitchToMedia_(selectedMedia)) {
```

Comment:
This is the only change that isn't behind a feature switch, but I think it is something that we should have been doing. This prevents us from switching aborting a bunch of things and then re-requesting the same playlist and requesting them again.